packages/flutter/test/widgets/editable_text_test.dart

```diff
@@ -1398,6 +1398,46 @@ void main() {
     }
   });
 
+  testWidgets('Read-only fields do not format text', (WidgetTester tester) async {
+    SelectionChangedCause selectionCause;
+
+    final TextEditingController controller =
+        TextEditingController(text: 'Lorem ipsum dolor sit amet');
+
+    await tester.pumpWidget(
+      MaterialApp(
+        home: EditableText(
+          readOnly: true,
+          controller: controller,
+          backgroundCursorColor: Colors.grey,
+          focusNode: focusNode,
+          style: textStyle,
+          cursorColor: cursorColor,
+          selectionControls: materialTextSelectionControls,
+          onSelectionChanged: (TextSelection selection, SelectionChangedCause cause) {
+            selectionCause = cause;
+          },
+        ),
+      ),
+    );
+
+    // Interact with the field to establish the input connection.
+    final Offset topLeft = tester.getTopLeft(find.byType(EditableText));
+    await tester.tapAt(topLeft + const Offset(0.0, 5.0));
+    await tester.pump();
+
+    expect(tester.testTextInput.hasAnyClients, kIsWeb ? isTrue : isFalse);
+    if (kIsWeb) {
+      tester.testTextInput.updateEditingValue(const TextEditingValue(
+        text: 'Foo bar',
+        selection: TextSelection(baseOffset: 0, extentOffset: 3),
+      ));
+      // On web, the only way a text field can be updated from the engine is if
+      // a keyboard is used.
+      expect(selectionCause, SelectionChangedCause.keyboard);
```

Comment:
should we also verify the handle does not show up?

Reply:
how do we verify it? can we add a screenshot test here?

Reply:
Something like this should be sufficient https://github.com/flutter/flutter/blob/d3fd62c8433ab9bf4e6b7312096cea1ab98b543f/packages/flutter/test/material/text_field_test.dart#L1032

Reply:
The thing is, `EditableText` doesn't show the handles on its own. Instead, it takes a `showSelectionHandles` property. This test is only making sure that `EditableText` reports the change correctly. The rest is on `SelectableText` and `material/TextField` to take the selection change and pass the correct value for `showSelectionHandles` (which they already do, and they have tests for that).

Reply:
Maybe I'll add a test that includes the entire interaction between `EditableText` and `material/TextField`?

Reply:
I think you can just use textfield/selectabletext to test directly, after all, we care about they do not show the handle but not how editabletext ineract with textfield to make that happen.

Reply:
Thanks for the suggestions! I wrote a test for `selectable_text` and `material/text_field` to make sure they don't show handles under these circumstances.

@chunhtai do you think I should remove the `editable_text` test that I had initially?

Reply:
I think we can keep it